sheetstorm/management/commands/generate_test_data.py

```diff
@@ -170,6 +218,144 @@ def _create_start_date(time_delta: Any = PROJECT_START_DATE_TIME_DELTA) -> timez
     def _create_stop_date(time_delta: Any = PROJECT_STOP_DATE_TIME_DELTA) -> timezone.datetime:
         return timezone.now() - time_delta
 
+    def execute_adding_users_to_projects(self) -> None:
```

Comment:
I was thinking how to improve this method because it calls 9 times same method. I figured out such solution:
```python
def _parse_data_to_add_users_to_projects(self):
    amount_of_users_map = {
        CustomUser.UserType.EMPLOYEE.name: {
            ProjectType.SUSPENDED.name: self.max_number_of_employees_in_suspended_project,
            ProjectType.ACTIVE.name: self.max_number_of_employees_in_active_project,
            ProjectType.COMPLETED.name: self.max_number_of_employees_in_completed_project,
        },
        CustomUser.UserType.ADMIN.name: {
            ProjectType.SUSPENDED.name: self.max_number_of_admins_in_suspended_project,
            ProjectType.ACTIVE.name: self.max_number_of_admins_in_active_project,
            ProjectType.COMPLETED.name: self.max_number_of_admins_in_completed_project,
        },
        CustomUser.UserType.MANAGER.name: {
            ProjectType.SUSPENDED.name: self.max_number_of_managers_in_suspended_project,
            ProjectType.ACTIVE.name: self.max_number_of_managers_in_active_project,
            ProjectType.COMPLETED.name: self.max_number_of_managers_in_completed_project,
        },

    }
    user_lists_map = {
        CustomUser.UserType.EMPLOYEE.name: self.employees_list,
        CustomUser.UserType.ADMIN.name: self.admins_list,
        CustomUser.UserType.MANAGER.name: self.managers_list,
    }
    project_lists_map = {
        ProjectType.SUSPENDED.name: self.suspended_projects_list,
        ProjectType.ACTIVE.name: self.active_projects_list,
        ProjectType.COMPLETED.name: self.completed_projects_list,
    }
    return (amount_of_users_map, user_lists_map, project_lists_map)

def execute_adding_users_to_projects(self) -> None:
    (amount_of_users_map, user_lists_map, project_lists_map) = self._parse_data_to_add_users_to_projects()

    for user_type in user_lists_map:
        for project_type in project_lists_map:
            self.add_users_to_projects(
                user_lists_map[user_type],
                project_lists_map[project_type],
                user_type,
                project_type,
                amount_of_users_map[user_type][project_type]
            )

    for user_type in user_lists_map:
        self.remove_all_random_user_projects(user_type)
```

Please take a look at it and consider using it or comment/improve etc

Reply:
Done. I implemented your solution